models/src/it/java/io/sphere/sdk/reviews/ReviewIntegrationTest.java

```diff
@@ -0,0 +1,392 @@
+    public void testCreate_InvalidProductId() throws Exception {
+        withCustomer(client(), customer -> {
+            for (int i = 0; i < REVIEWS.length; i++) {
+                Review review = client().execute(new ReviewCreateCommand(ReviewDraft.of("invalid", customer.getId())
+                        .withAuthorName((String) REVIEWS[i][0]).withTitle((String) REVIEWS[i][1])
+                        .withText((String) REVIEWS[i][2]).withScore((Float) REVIEWS[i][3])));
+            }
+        });
+    }
+
+    @Test(expected = io.sphere.sdk.client.ErrorResponseException.class)
+    public void testCreate_NullCustomer() throws Exception {
+        withProduct(client(), product -> {
+            for (int i = 0; i < REVIEWS.length; i++) {
+                Review review = client().execute(new ReviewCreateCommand(ReviewDraft.of(product.getId(), null)
+                        .withAuthorName((String) REVIEWS[i][0]).withTitle((String) REVIEWS[i][1])
+                        .withText((String) REVIEWS[i][2]).withScore((Float) REVIEWS[i][3])));
+            }
+        });
+    }
+
+    @Test(expected = io.sphere.sdk.client.ErrorResponseException.class)
+    public void testCreate_InvalidCustomerId() throws Exception {
+        withProduct(client(), product -> {
+            for (int i = 0; i < REVIEWS.length; i++) {
+                Review review = client().execute(new ReviewCreateCommand(ReviewDraft.of(product.getId(), "invalid")
+                        .withAuthorName((String) REVIEWS[i][0]).withTitle((String) REVIEWS[i][1])
+                        .withText((String) REVIEWS[i][2]).withScore((Float) REVIEWS[i][3])));
+            }
+        });
+    }
+
+    @Test(expected = io.sphere.sdk.client.ErrorResponseException.class)
+    public void testCreate_ConstraintNoTitleTextScore() throws Exception {
+        withProduct(client(), product ->
+            withCustomer(client(), customer -> {
+                Review review = client().execute(new ReviewCreateCommand(ReviewDraft.of(product.getId(), customer.getId())
+                        .withAuthorName((String) REVIEWS[0][0])));
+            }));
+    }
+
+    @Test(expected = io.sphere.sdk.client.ErrorResponseException.class)
+    public void testUpdateNoTitleTextScore() throws Exception {
+        withProduct(client(), product ->
+                withCustomer(client(), customer ->
+                        withReview(client(), product, customer, review -> {
+                            client().execute(ReviewUpdateCommand.of(
+                                    Versioned.of(review.getId(), review.getVersion()), Arrays.asList(
+                                            SetTitle.of(Optional.empty()),
+                                            SetText.of(Optional.empty()),
+                                            SetScore.of(Optional.empty()))));
+                        })));
+    }
+
+    @Test
+    public void testCreate_ConstraintTitleTextScore() throws Exception {
```

Comment:
I think one test with leaving out all optional values and one test with all possible filled fields is sufficient.